src/trio/_core/_run.py

```diff
@@ -744,6 +756,25 @@ def deadline(self, new_deadline: float) -> None:
         with self._might_change_registered_deadline():
             self._deadline = float(new_deadline)
 
+    @property
+    def relative_deadline(self) -> float | None:
+        """TODO: write docstring"""
+        if self._has_been_entered:
+            if self.deadline == inf:
+                return None
+            return self._deadline - current_time()
+        return self._relative_deadline
+
+    @relative_deadline.setter
+    def relative_deadline(self, relative_deadline: float | None) -> None:
+        self._relative_deadline = relative_deadline
+        if self._has_been_entered:
+            with self._might_change_registered_deadline():
+                if relative_deadline is None:
+                    self._deadline = inf
```

Comment:
I think this is confusing in conjunction with allowing people to set both `deadline` and `relative_deadline`.